python/dcr_template.py

```diff
@@ -241,55 +548,68 @@ def _create_exposure(self, array, variance=None, elevation=None, azimuth=None, s
         meta.add("AIRMASS", 1.0/np.sin(np.radians(elevation)))
         meta.add("ZENITH", 90 - elevation)
         meta.add("AZIMUTH", azimuth)
+        # Add all additional keyword arguments to the metadata.
         for add_item in kwargs:
             meta.add(add_item, kwargs[add_item])
-        return(exposure)
+        return exposure
 
     def export_model(self, model_repository=None):
+        """Persist a DcrModel with metadata to a repository.
+
+        @param model_repository: full path to the directory of the repository to save the dcrModel in.
+        """
         if model_repository is None:
-            butler_model = self.butler
+            butler = self.butler
         else:
-            butler_model = daf_persistence.Butler(model_repository)
-        wave_gen = _wavelength_iterator(self.bandpass, use_midpoint=False)
-        for _f in range(self.n_step):
+            butler = daf_persistence.Butler(model_repository)
+        wave_gen = DcrModel._wavelength_iterator(self.bandpass, use_midpoint=False)
+        for f in range(self.n_step):
             wl_start, wl_end = wave_gen.next()
-            exp = self._create_exposure(self.model[_f, :, :], variance=self.weights[_f, :, :],
+            exp = self._create_exposure(self.model[f, :, :], variance=self.weights[f, :, :],
                                         elevation=90., azimuth=0., ksupport=self.kernel_size,
-                                        subfilter=_f, nstep=self.n_step, wavelow=wl_start, wavehigh=wl_end,
+                                        subfilt=f, nstep=self.n_step, wavelow=wl_start, wavehigh=wl_end,
                                         wavestep=self.bandpass.wavelen_step, psf_flag=self.use_psf)
-            butler_model.put(exp, "dcrModel", dataId=self._build_model_dataId(self.photoParams.bandpass, _f))
+            butler.put(exp, "dcrModel", dataId=self._build_model_dataId(self.photoParams.bandpass, f))
 
     def load_model(self, model_repository=None, band_name='g', **kwargs):
+        """Depersist a DcrModel from a repository and set up the metadata.
```

Comment:
Yes, the __init__ mostly calls this function but I wanted other classes to be able to inherit the ability to load models.